Lib/multiprocessing/connection.py

```diff
@@ -400,11 +400,12 @@ def _recv(self, size, read=_read):
         buf = io.BytesIO()
         handle = self._handle
         remaining = size
-        is_pipe = False
+        is_pipe = is_socket = False
         if size > self._default_pipe_size > 0:
             mode = os.fstat(handle).st_mode
             is_pipe = stat.S_ISFIFO(mode)
-        limit = self._default_pipe_size if is_pipe else remaining
+            is_socket = stat.S_ISSOCK(mode)
+        limit = self._default_pipe_size if is_pipe or is_socket else remaining
```

Comment:
How about use the limit for all connection, instead of socket and pipe?

Reply:
If there are no performance drawbacks for other types of connections, then it's a solid solution. I've only analyzed pipes and sockets.

Reply:
I don't know connection other than pipe and socket.
But mmap+mremap+munmap overhead is very common issue for all streams.
So using chunked read by default and override it in special connection class seems better solution.